Intermediates (#184)
ihmeuw/vivarium_research: docs/source/model_design/risk_factors.rst

Intermediates
"""""""""""""

An intermediate variable as discussed in this section is defined as a variable 
between an exposure and outcome in a sequential causal chain, as demonstrated 
in the diagram below:

.. image:: basic_int_diagram.svg

A (simplified) example that can demonstrate such a causal pathway is the 
relationship between the exposure of cigarette smoking, which is causally 
related to the intermediate variable of accumulated tar in lungs, which in 
turn is causally related to lung cancer (note: for the purposes of this 
example, assume that there is no direct causal relationship between cigarette 
smoking and lung cancer other than through the intermediate variable of 
accumulated tar in lungs).

.. image:: smoking_intermediate_example.svg

Now, let's say that we plan to enact an advertising campaign to reduce smoking 
in hopes of reducing population lung cancer rates. Our expected causal pathway 
would look like this:

.. image:: smoking_intervention_example.svg

Therefore, as we increase intervention coverage, we would exect a decrease in 
population lung cancer rates, as these two variables are located on the same 
causal pathway. 

> However, let's imagine that a brand new hypothetical medication was just 

Reviewer: You could perhaps give headings to your example sections. This feels like a new type of example, because before, smoking was a proxy for tar in lungs, and now, it's a poor measure of tar in lungs without information on medication coverage.

Reviewer: Also-- you had mentioned/asked for suggestions on alternatives examples of intermediates. I'm thinking about categories:

1.  good proxies (maybe smoking/tar in lungs)
2.  worse proxies but still known direct relationship (smoking/tar in lungs, despite existence of medication)
3. assumed or observed proxies?

I'm not sure if in #3 you leave mediation. For example, if you have data on years of education, and you know this is linked to income, and you want to model some outcome for which income is a driver of change---could you treat and model this as mediation, even if perhaps the relationship is quite a bit more complicated?

This seems like it should be disambiguated somewhere, but you suggested during coworking that perhaps that occurs after all of these causal relationship sections.

Author: Thanks for the suggestions, Beatrix! I added some sub-headings and agree that they make the section more read-able. So far, I included "An Example of Uncomplicated Intermediates", "An Example of Complicated Intermediates", and "Other Considerations". Let me know if you think these could be improved!

As for your point about potential overlap between issues like intermediates and mediation, I'm thinking that it might be good to include a discussion of this in a section called "Causal Inference and the Simulation Science Team" (similar to the section "Causal Inference in the Global Burden of Disease Study" that was added in PR#201). where we talk about how we might consider how we might be limited by data gaps in applying our understanding of these concepts (defined in their individual sections) to specific interventions or risks and how such data gaps might affect the certainty in our model results. Let me know how this sounds. 

Thanks :)

relationship between smoking and lung cancer in a simulation for a different 
location. However, in the location we wish to model, the medication is not 
approved at a national level and therefore use of the medication is close to 
zero. However, since we know that the relationship between smoking and lung 
cancer as we've defined it in this example is affected by the use of this 
medication, applying the data from this new study to this model location would 
be inappropriate. Rather, we should use data that measured the relationship 
between smoking and lung cancer in a study population with a similar exposure 
to the medication intermediate variable.

Another example of when an intermediate variable might interfere with the 
relationship between an exposure and outcome is when a given intervention 
*decreases* the prevalence of an intermediate variable (that is related to an 
outcome), but the prevalence of that intermediate variable in a specific 
location is already zero and therefore cannot be decreased any further. The 
opposite example of an intervention that *increases* the prevalence of an 
intermediate variable that is already 100% prevalent also holds true.

> The impact of interference by intermediate variables between a given exposure 

Reviewer: I think this section could also perhaps have its own subheading!

Intermediates
"""""""""""""

An intermediate variable as discussed in this section is defined as a variable 
between an exposure and outcome in a sequential causal chain, as demonstrated 
in the diagram below:

.. image:: basic_int_diagram.svg

An Example of an Uncomplicated Intermediate
*******************************************

A (simplified) example that can demonstrate such a causal pathway is the 
relationship between the exposure of cigarette smoking, which is causally 
related to the intermediate variable of accumulated tar in lungs, which in 
turn is causally related to lung cancer (note: for the purposes of this 
example, assume that there is no direct causal relationship between cigarette 
smoking and lung cancer other than through the intermediate variable of 
accumulated tar in lungs).

> 

Author: Yes, I like this better. Will update!

.. image:: smoking_intermediate_example.svg

Now, let's say that we plan to enact an advertising campaign to reduce smoking 
in hopes of reducing population lung cancer rates. Our expected causal pathway 
would look like this:

.. image:: smoking_intervention_example.svg

Therefore, as we increase intervention coverage, we would exect a decrease in 
population lung cancer rates, as these two variables are located on the same 
causal pathway. 

An Example of a Complicated Intermediate
****************************************

Now, let's imagine that a brand new hypothetical medication was just 
created that dissolves tar in lungs that accumulates due to smoking and is now 
widely used in some areas. Our causal diagram would now look like this:

.. image:: medication_example.svg

Quickly, we can now see that the relationship we previously knew between 
smoking and lung cancer is now impacted by the use of this hypothetical 
medication that affects the intermediate variable between smoking and lung 
cancer. Therefore, now we can only expect our intervention to decrease 
population lung cancer rates only in areas that do not widely use this 

> medication.

Author: Yes, thanks. I will clarify this.

Notably, intermediate variables may be relevant in situations in which the 
variable located most proximal to the outcome in the causal pathway is 
difficult to measure. For instance, measuring the amount of tar in a person's 
lungs is an invasive procedure; measuring the amount of cigarettes a person 
smokes is much easier in comparison. Therefore, data availability may dictate 

> that we model cigarette smoking rather than lung tar. 

Author: Done :)

  This may be a reasonable approach, given that the intermediate variable 
  *does not* interfere with the relationship between the exposure and outcome 
  for the given model population (little or no use of the medication that 

>   removes tar from lung). However, if there *is* interference by the 

Author: You're right! Thanks, will update

  intermediate variable on the relationship between the exposure and outcome 

>   (significant use of the medication), then the previously measured 

Author: I think this is important to keep because there still will be *some* association between cancer and smoking (even if it is zero), but it will not be the same as what it was when it was measured in the absence of the medication.

  relationship between smoking and lung cancer will not apply to this 
  population in the same way.

Now, let's say that some time has gone by and now this medication has become 
quite common. A new study measured the relationship between smoking and lung 
cancer in a study population in which 50% of smokers used the medication. Now, 
let's say that we are interested in using the data from that study on the 
relationship between smoking and lung cancer in a simulation for a different 
location. However, in the location we wish to model, the medication is not 
approved at a national level and therefore use of the medication is close to 
zero. However, since we know that the relationship between smoking and lung 
cancer as we've defined it in this example is affected by the use of this 
medication, applying the data from this new study to this model location would 
be inappropriate. Rather, we should use data that measured the relationship 
between smoking and lung cancer in a study population with a similar exposure 

> to the medication intermediate variable.

Reviewer 2: I think here, we can be a bit more explicit about what the intermediate variable is which is tar in the lungs. The medication is an intervention that affects the intermediate variable. 

Maybe say: 
we should use data that measured the relationship between smoking and lung cancer in a study population with a similar exposure the medication which determines the level of the intermediate variable.

Author: Agreed, thanks

Additional Considerations
*************************

Another example of when an intermediate variable might interfere with the 
relationship between an exposure and outcome is when a given intervention 

> *decreases* the prevalence of an intermediate variable (that is related to an 

Reviewer 2: isnt this the medication that dissolves tar example?

Author: Yeah, I guess I was thinking that this example could also be that there is just no baseline burden of the intermediate variable (like no malaria or zika exposure because of climate or something along those lines). Do you think this should be clarified and/or that it is worthwhile to include?

Author: (See my response in the comment thread below)

outcome), but the prevalence of that intermediate variable in a specific 

> location is already zero and therefore cannot be decreased any further. The 

Reviewer 2: if the prevalence of the intermediate var is 0, then how does our main variable act through the intermediate variable to outcome on the causal pathway?

Author: So this would be something like what happened regarding NTDs in LSFF...

Where 
exposure = folic acid fortification
intermediate = folic acid deficiency
outcome = neural tube defects

folic acid fortification decreases folic acid deficiency which causes neural tube defects
(however, neural tube defects might be caused by other things as well)

So we would expect increasing folic acid fortification to decrease neural tube defects, but if there is no folic acid deficiency in the population, then it wouldn't actually have an effect.

Maybe this needs more clarification?

Author: I added this as an example in the latest commit :)

Reviewer 2: ah! this is much more clarified thank you! so essentially if it just means that if the prevalence of that intermediate variable is 0, our main exposure cannot act through that causal pathway anymore.

Author: Yes, exactly!

  between the exposure and outcome (i.e. significant use of the medication), 
  then the previously measured relationship between smoking and lung cancer 
  will not apply to this population in the same way.

Now, let's say that some time has gone by and now this medication has become 
quite common. A new study measured the relationship between smoking and lung 
cancer in a study population in which 50% of smokers used the medication. Now, 
let's say that we are interested in using the data from that study on the 
relationship between smoking and lung cancer in a simulation for a different 
location. However, in the location we wish to model, the medication is not 
approved at a national level and therefore use of the medication is close to 
zero. However, since we know that the relationship between smoking and lung 
cancer as we've defined it in this example is affected by the use of this 
medication, applying the data from this new study to this model location would 
be inappropriate. Rather, we should use data that measured the relationship 
between smoking and lung cancer in a study population with a similar exposure 

> to the medication intermediate variable.

Reviewer 2: ```suggestion
to the exogenous variable (medication) on the intermediate variable.
```

Additional Considerations
*************************

Another example of when an intermediate variable might interfere with the 
relationship between an exposure and outcome is when a given intervention 
*decreases* the prevalence of an intermediate variable (that is related to an 
outcome), but the prevalence of that intermediate variable in a specific 
location is already zero and therefore cannot be decreased any further. The 
opposite example of an intervention that *increases* the prevalence of an 
intermediate variable that is already 100% prevalent also holds true.

	For instance, imagine the example of folic acid supplementation 
	(exposure), which *decreases* folic acid deficiency (intermediate), which
	causes neural tube birth defects (outcome). Notably, neural tube birth 

> 	defects are also caused by exogenous variables such as maternal diabetes. 

Reviewer 2: ```suggestion
	defects are also caused by other causal pathways such as maternal diabetes. 
```

> 	Given that this causal pathway is true, we would expect an increase in 

Reviewer 2: ```suggestion
	Given that the causal pathway from folic acid fortification-->folic acid deficiency-->neural tube defects is true, we would expect an increase in 
```

	exposure to folic acid supplementation to decrease neural tube defects. 
	However, the maximum effect of increasing exposure to folic acid 

> 	supplementation is dependent on the prevalence of folic acid deficiency in 

Reviewer 2: ```suggestion
	supplementation is dependent on the prevalence of the intermediary, folic acid deficiency, in 
```

> 	the population. Notably, if folic acid deficiency is zero, the effect of 

Reviewer 2: ```suggestion
	the population. Notably, if folic acid deficiency is zero,
```

> 	increasing folic acid supplementation exposure will also be zero.

Reviewer 2: ```suggestion
	increasing folic acid supplementation exposure will have no effect on neural tube defects.
```

example, assume that there is no direct causal relationship between cigarette 
smoking and lung cancer other than through the intermediate variable of 
accumulated tar in lungs).

.. image:: smoking_intermediate_example.svg

Now, let's say that we plan to enact an advertising campaign to reduce smoking 
in hopes of reducing population lung cancer rates. Our expected causal pathway 
would look like this:

.. image:: smoking_intervention_example.svg

Therefore, as we increase intervention coverage, we would exect a decrease in 
population lung cancer rates, as these two variables are located on the same 
causal pathway. 

However, let's imagine that a brand new hypothetical medication was just 
created that dissolves tar in lungs that accumulates due to smoking and is now 
widely used in some areas. Our causal diagram would now look like this:

.. image:: medication_example.svg

Quickly, we can now see that the relationship we previously knew between 
smoking and lung cancer is now impacted by the use of this hypothetical 
medication that affects the intermediate variable between smoking and lung 
cancer. Therefore, now we can only expect our intervention to decrease 

> population lung cancer rates only in areas that do widely use this medication.

Reviewer: do not

Author: Thank you!!!
